ctapipe/io/tableloader.py

```diff
@@ -119,9 +134,173 @@ def _read_telescope_table(self, group, tel_id):
+        
+            try:
+                for tel_type in output.keys():
+                    output[tel_type].sort(sort_by)
+            except ValueError:
+                for tel_type in output.keys():
+                    output[tel_type].sort(["tel_id"])
+        
+        self.close()
+        
+        return output
+    
+    def read_telescope_events_by_tel_type(self, tel_ids=None):
+                
+        selected_subarray = self.subarray.select_subarray(tel_ids, name="")
+        selected_tel_types = selected_subarray.telescope_types
+        
+        table_dict = {}
+        for tel_type in selected_tel_types:
+            
+            selected_tel_ids = [tel_id for tel_id in tel_ids if tel_id in self.subarray.get_tel_ids_for_type(tel_type)]
+            table_dict[str(tel_type)] = self.read_telescope_events(selected_tel_ids)
+        
+        return table_dict
+
+    def read_telescope_events(self, tel_ids=None):
+        if tel_ids is None:
+            tel_ids = self.subarray.tel.keys()
+        
         table = None
+        
+        # Joini all the DL1-based data, which depends on the triggered tel_ids
+        if (self.load_dl1_images or self.load_dl1_parameters or self.load_true_images) and ((table is None) or (len(table)==0)):
+            table = vstack([self.read_telescope_events_for_id(tel_id) for tel_id in tel_ids])
+        elif (self.load_dl1_images or self.load_dl1_parameters or self.load_true_images) and (len(table) > 0):
+            table = join(
+                table,
+                vstack([self.read_telescope_events_for_id(tel_id, table) for tel_id in tel_ids]),
+                keys=["tel_id"],
+                join_type="inner",
+            )
+        
+        # Instrument information doesn't depend on the triggered tel_ids
+        # but depends on the tel_ids for joining
+        if self.load_instrument and ((table is None) or (len(table)==0)):
+            table = self.instrument_table
+        elif self.load_instrument and len(table) > 0:
+            table = join(
+                table, self.instrument_table, keys=["tel_id"], join_type="inner"
+            )
+        
+        # DL2 data (either simulated or reconstructed) depends only on
+        # event_id(s), not on any specific tel_id(s)
+        # so we add it at the end
+        if self.load_dl2_geometry:
+            
+            # get reconstructors in file (if more than one)
+            reconstructors = self._get_reconstructors()
+                        
+            for i, reconstructor in enumerate(reconstructors):
+                
+                geometry = self._read_telescope_DL2_geometry(GEOMETRY_GROUP, reconstructor)
+            
+                # rename DL2 columns to explicit reconstructor
+                # TBD: we could skip this if only 1 reconstructor is present
+                # or simply find another way to deal with multiple reconstructions
```

Comment:
I think it's ok to always prepend the reconstructor name.  That way there is no confusion about which information is being used.   In the DL2→DL3 transition, we reduce this to a single reconstruction, but not in DL2 where all tools should assume there are many.

Reply:
if we keep it like this I guess that in the next tool we need to strip somewhat these prefixes or add a dictionary of supported reconstructions to read them